### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by SiahaanBernard**

### CODE REVIEW

File: data.tf [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      "ssmmessages:OpenDataChannel",
+    ]
+
+    resources = [
+      "*",
+    ]
+  }
+
+  statement = {
+    sid    = "AllowEC2InstancesToGetBucketEcnryption"
```
Comment (febryantonius) [inline data.tf]: minor typo `Ecnryption`

File: examples/main.tf [hunk 2]
```diff
@@ -0,0 +1,7 @@
+module "session_manager_config" {
+  source                   = "../"
+  product_domain           = "tsi"
+  environment              = "staging"
+  s3_bucket_description    = "s3 bucket to store session manager logs"
```
Comment (febryantonius) [inline examples/main.tf]: I'm just wondering is it necessary to set these (`s3_bucket_description` and `ssm_document_description`) as required variable?

Comment (SiahaanBernard) [reply examples/main.tf]: Hm this is used for tagging the resources, to follow our tagging policy. Do you think this is something that should be generated?

File: data.tf (hunk 1 above)
Comment (SiahaanBernard) [reply data.tf]: ah ok.

**Pull request merged**